Failing tests on M1-Arm64 on develop

<!---
Thank you for reporting a problem and suggesting improvements. Please provide the below information to make sure we can effectively deal with the issue reported. For the most precise version information, see the first line of console output or run with --version. Please attach or include example code that allows us to reproduce the problem.
--->

CBMC version: 5.23.0 (cbmc-5.21.3-exp-254-g268f8864e)
Operating system: macOS 11.0
Exact command line resulting in the issue: `cbmc/regression/ansi-c/gcc_version1 $ clang gcc-5.c`
What behaviour did you expect: Binary is compiled without issue.
What happened instead: Clang fails with error: `__float128 is not supported on this target`.

---

Running the regression test-suite under make on macOS Big Sur, on top of the new m1 platform is causing some of the regression tests to fail. Specifically, `regression/ansi-c/gcc-version1` and `regression/ansi-c/gcc-version2` are failing.

On closer inspection, invoking clang directly on the binary gives me this:

```sh
fotiskoutoulakis@Fotiss-Mac-mini gcc_version1 % clang gcc-5.c
gcc-5.c:10:1: error: __float128 is not supported on this target
__float128 f128;
^
```

Querying clang about version and target produces this:

```sh
fotiskoutoulakis@Fotiss-Mac-mini gcc_version1 % clang -v 
Apple clang version 12.0.0 (clang-1200.0.32.27)
Target: arm64-apple-darwin20.1.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

I have two questions/remarks:

* It seems like it is assumed that `__float128` is going to be supported on other platforms, but this looks like an unsafe assumption. Should we change the test/deactivate it on some platforms?
* It looks like a subset of the tests in `regression/ansi-c` are marked `gcc-only`. Do we need to run them under MacOS where the default compiler is going to be `clang/llvm`? If not we may need to make changes to our CI infrastructure, as we currently *do* run them under the `macOS-cmake` and `macOS-make` builds.

--- a/regression/ansi-c/gcc_version1/gcc-4.c
+++ b/regression/ansi-c/gcc_version1/gcc-4.c
@@ -1,4 +1,5 @@
-// None of these types should be defined when emulating gcc-4:
+// None of these types should be defined when emulating gcc-4, which clang on
+// macOS also emulates:
 
 typedef float _Float32;
 typedef double _Float32x;

--- a/regression/ansi-c/gcc_version1/gcc-5.c
+++ b/regression/ansi-c/gcc_version1/gcc-5.c
@@ -1,4 +1,6 @@
 // These types should *not* be provided when emulating gcc-5:
+
+#ifndef __clang__
 typedef float _Float32;
 typedef double _Float32x;
 typedef double _Float64;
@@ -8,3 +10,4 @@ typedef long double _Float128x;
 
 // But this type should:
 __float128 f128;
+#endif

--- a/regression/ansi-c/gcc_version1/gcc-7.c
+++ b/regression/ansi-c/gcc_version1/gcc-7.c
@@ -1,4 +1,6 @@
 // All these types should be provided when emulating gcc-7:
+
+#ifndef __clang__
 _Float32 f32;
 _Float32x f32x;
 _Float64 f64;
@@ -7,3 +9,4 @@ _Float128 f128;
 _Float128x f128x;
 
 __float128 gcc_f128;
+#endif

